app/controllers/actions/AuthAction.scala

```diff
@@ -142,6 +148,28 @@ class AuthActionImpl @Inject() (
       throw NoExternalId()
     }
   }
+
+  def allowListAuthentication(eori: String): Boolean = {
+    val eoriOnAllowList = eoriAllowList.allows(eori)
+
+    if (!eoriOnAllowList)
+      logger.info("Authentication Rejected: User's EORI not on allow list")
+
+    eoriOnAllowList
+  }
+
+  def tdrSecretAuthentication(eori: String, maybeHiddenSalt: Option[String], providedHash: String): Boolean =
```

Comment:
Nit: I don't like the prefix `maybe` when we can see that in the type, feels redundant

Reply:
fussy :smile:

Reply:
Sorry if I disagree with you Tom :blush: I find it useful because then in higher-order functions you can then use `hiddenSalt =>`.
Less important but still to point out it's quite an accepted pattern in the FP community.

Reply:
fair enough, I guess the type isn't always explicit